prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirTransformer.kt

```diff
@@ -213,7 +219,11 @@ class FhirTransformer(
         }
         // Finally set the value
         val property = childResource.getNamedProperty(pathParts.last())
-        val newValue = FhirBundleUtils.convertFhirType(value, value.fhirType(), property.typeCode)
+        if (property == null) {
+            logger.warn("Could not find property '${pathParts.last()}'.")
+            return
```

Comment:
Too are many returns in this function. Is there a way to refactor this?

Reply:
I extracted some of the validation at the beginning of the function into a helper so there is one return there instead of three, and then reordered some logic to remove another one. Hopefully that helps. Will check that in with a new commit shortly.